--- /dev/null
+++ ../src/Application/ApplicationFileProcessor.php
@@ -259,11 +259,11 @@
         if (!isset($_SERVER[self::ARGV][0])) {
             return null;
         }
+        var_dump($_SERVER[self::ARGV);
         $potentialEcsBinaryPath = $_SERVER[self::ARGV][0];
         if (!\file_exists($potentialEcsBinaryPath)) {
             return null;
         }
         return $potentialEcsBinaryPath;
     }
     /**
      * @param File[] $files
